synapse/push/clientformat.py

```diff
@@ -41,13 +41,6 @@ def format_push_rules_for_user(
 
         rulearray.append(template_rule)
 
-        for type_key in ("pattern", "value"):
```

Comment:
I removed this since I don't think this can happen at the root level.

Reply:
Right so the rule using this has been [deprecated](https://spec.matrix.org/v1.7/client-server-api/#default-content-rules) in v1.7 of the spec.

Reply:
But I think we need to keep it:
https://github.com/matrix-org/synapse/blob/develop/synapse/push/clientformat.py#L104